tests/foreman/api/test_repository.py

```diff
@@ -47,7 +47,10 @@ def repo_options(request, module_org, module_product):
     """Return the options that were passed as indirect parameters."""
     options = getattr(request, 'param', {}).copy()
     options['organization'] = module_org
-    options['product'] = module_product
+    if module_product:
+        options['product'] = module_product
+    else:
+        options['product'] = entities.Product(organization=module_org).create()
```

Comment:
module_product fixture will create a product anyway, so why we need this else condition?

Reply:
So if module_product is deleted by another test, it will be recreated before the next one?

Reply:
I see the things here module product is referenced with the repo_options fixture and is being deleted from one of the test. Causing many failures. 

The current solution will work as it will create the new product in the case of the repo_options fixture.

Another solution would be just before deleting the product call use another product rather than module_product.

Reply:
- With use of fixture `repo_options_custom_product` inside test function `test_positive_delete_product_with_synced_repo` will make sure delete the custom created product.
- Verified in target satellite web-ui
- With this changes test `TestDockerRepository::test_positive_update_name` will not failed due to error
- Reference - [satqe-jenkins-csb-satellite-qe.apps.ocp-c1.prod.psi.redhat.com -> TestDockerRepository/test_positive_update_name_docker](https://satqe-jenkins-csb-satellite-qe.apps.ocp-c1.prod.psi.redhat.com/view/6.12%20Components/job/sat-6.12-rhel8-Repositories/29/testReport/junit/tests.foreman.api.test_repository/TestDockerRepository/test_positive_update_name_docker_/)

Reply:
discard above changes, use existing fixture which creates custom product.
Hence marking this conversation as resolve.